api/util/flask.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any
+
+from flask import Flask
+from sqlalchemy.orm import Session
+
+
+class PalaceFlask(Flask):
+    """
+    A subclass of Flask sets properties used by Palace.
+
+                       ╒▓▓@=:===:╗φ
+                        ,██Ü_╠░_█▌_
+             ___     ─-'██▌  ▌H ▐█▓
+      ______░╓___________▐█▌▄╬▄▄█____________________
+     ▐██▒         `````░░░░░Ü░░░░╠╠╠╠╬╬╬╬╬╬╬╬╠▒▒Ü` ▐▓L
+     ▐██▌              »»»░░░░░░░▒▒╠╠╠╬╬╬╬╬╬╠▒▒ÜÜ  ╚▓▒
+     ▐██▌              `»│»»»░░░░▒▒▒╠╠╬╬╬╬╬╬╠▒▒░░  ╚▓▌
+     ▐██▌              »»»»»░░░░░▒▒▒╠╠╬╬╬╬╬╬╠▒ÜÜ░  ▐▓▌
+     ▐██▌             `»»»»░░░░░░▒▒▒╠╠╬╬╬╬╬╬▒▒Ü░░  ▐▓▌
+     ▐█▓▌              `»»»»░░░░░▒▒▒╠╠╠╬╬╬╬╬▒▒Ü░░  ╚▓▌
+     ▐█▓▌              »»»»░░░░░░▒▒▒╠╠╠╠╬╬╬╠Ü▒░░░  ╠▓▌
+     ▐█▓▌              »»»»░░░░░░▒▒▒▒╠╠╠╬╬╬╠▒Ü░░░  ╠╣▌
+     ▐██▌                               ╬╬╬╠▒▒░░░  ╠╣▌
+     ▐██▌           The Palace Project       ▒░░░  ╠╣▌
+     ▐██▌                               ╠╬╠╠▒▒░░░  ║╣▌
+     ▐██▌              »»»»░░░░░░▒▒▒▒▒╠╠╠╠╠╠▒▒░░░  ║▓▌
+     ▐██▌             `»»»░░░░░░░░▒▒▒▒╠╠╠╠╠▒▒▒░░░  ╠╣▌
+     ▐██▌             »»»»░░░░░░░░▒Ü▒▒╠╠╠╠╠▒▒Ü░░░  ▐╫▌
+     ▐██▌             `»»»░░░░░░░░▒▒▒▒╠╠╠╠╠▒▒Ü░░░  ▐╣▌
+      ██╬░     `»»»``»»»»░░░░░░░░░░▒▒▒▒▒▒▒▒▒▒Ü░░░░ [╣▌
+      █▓╬░     `»` »`»»»»»░░░░░░░░░░▒▒▒▒▒▒▒▒▒Ü░░░░ [╫▌
+      ▓▓╬░      `   `»»»»»»░░░░░░░Ü▒▒▒▒▒▒▒▒▒▒Ü░░░░ [╫▌
+      ▓▓╬░»``  » `  » »»»»░░░░░░░░▒▒▒▒▒▒▒▒▒▒▒Ü░░░░ [╫▌
+      ▓▓╬░»_` ``»`»»»»»»»»»░░░░░░Ü▒▒▒▒▒▒▒▒▒▒▒▒Ü░░░ |╫▌
+      ▓▓╬░»»»»»»»»»»»»»»»»░░░░░░░▒░▒▒▒▒▒▒▒▒▒▒ÜÜ░Ü░_|╟▌
+      ╣▓╬░»»»»»»»»»»»»»»»»»░░░░░░░▒▒▒▒▒▒▒▒▒▒▒▒▒ÜÜ░⌐|╟▌
+      ╝▓╬░__»»»»»»»»»»░»░░░░░░░░░▒▒▒▒▒▒▒▒▒▒▒▒▒▒░░░_|╠╛
+      ╚╚╚╙╙╙╙╙╙╙╙╙╙╙╙╙╙╙╙╙╙╙╙╙╚╚╚╚╚╚╚ÜÜÜÜÜÜÜÜÜÜÜÜÜ╙╚ÜH
+
+    Palace: You're going to need a stiff drink after this.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._db: Session
+        self.manager: Any
```

Comment:
Why not have this as a "CirculationManager"? It would make autocomplete work

Reply:
Thats a good point Rishi. I originally had this annotated as `CirculationManager`, but it caused other type errors, because the CirculationManager object doesn't have type hints for its members. 

The members of CirculationManager actually get setup in [setup_admin_controllers](https://github.com/ThePalaceProject/circulation/blob/f3c8210588107d7e5739160bdbc894b8fca0d4f9/api/admin/controller/__init__.py#L123-L129) instead of the __init__ function, so the type hints aren't implied.

So I was lazy and just made this `Any`. Maybe a better solution is to add type hints to `CirculationManager` so we actually get type info and autocomplete.

I'll see what it will take to do this.

Reply:
Okay, this should be addressed now. I added hints to `CirculationManager`.